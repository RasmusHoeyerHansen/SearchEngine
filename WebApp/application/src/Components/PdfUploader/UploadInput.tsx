import {IFileUploadFormProps} from "./FileUploadForm";

interface IFileInputProps extends IFileUploadFormProps{
    acceptedFileString? : string;
}



export type ValidFileTypes = 'application/pdf' | 'application/vnd.ms-excel';

export function FileInput ({acceptedFileString = "application/pdf", onChange}: IFileInputProps) : JSX.Element {
    return <input id={'FileUploadForm-input'}
                  type="file"
                  name="formFile" multiple
                  accept={acceptedFileString}
                  onChange={onChange}
                  role="input"
    />
}