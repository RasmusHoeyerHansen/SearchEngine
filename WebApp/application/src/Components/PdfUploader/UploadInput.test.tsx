import {cleanup, fireEvent, render, screen} from "@testing-library/react";
import React from "react";
import {FileInput} from "./UploadInput";

describe("FileInput", () => {
    beforeEach(() => {
        jest.clearAllMocks();
        cleanup();
    });
    it("renders correctly with default pdf as filetype and matches" +
        " snapshot", () => {
        let {container} = render(<FileInput onChange={() => {
            return;
        }}/>)
        expect(container).toMatchSnapshot();
    });


    it('should render an file input element', () => {
        render(<FileInput onChange={() => {
            return;
        }}/>);

        const inputElement = screen.getByRole("input") as HTMLInputElement;
        expect(inputElement).toBeTruthy();
        expect(inputElement.type).toBe('file');
        expect(inputElement.multiple).toBe(true);
    });

    it('should be able to render a pdf file', function () {
        render(<FileInput onChange={() => {
            return;
        }}/>);

        const file = new File([new ArrayBuffer(1)], 'testfile.pdf');
        let fileInput =  screen.getByRole("input") as HTMLInputElement;

        fireEvent.change(fileInput, {
            target: {files: [file]}
        });
        expect(fileInput.files).toBeTruthy();
        expect(fileInput.files?.length).toBeTruthy();
        expect(fileInput.files?.length).toBe(1);
    });
    it('should be able to render multiple pdf files', function () {
        render(<FileInput onChange={() => {
            return;
        }}/>);
        const file = new File([new ArrayBuffer(1)], 'testfile.pdf');
        const file2 = new File([new ArrayBuffer(1)], 'testfile2.pdf');
        const file3 = new File([new ArrayBuffer(1)], 'testfile3.pdf');

        let fileInput = screen.getByRole("input") as HTMLInputElement;
        fireEvent.change(fileInput, {
            target: {files: [file, file2, file3]}
        });

        expect(fileInput.files).toBeTruthy();
        expect(fileInput.files?.length).toBeTruthy();
        expect(fileInput.files?.length).toBe(3);
    });
})