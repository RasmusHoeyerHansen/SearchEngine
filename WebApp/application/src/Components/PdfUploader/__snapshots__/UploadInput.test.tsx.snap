// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput renders correctly with default pdf as filetype and matches snapshot 1`] = `
<div>
  <input
    accept="application/pdf"
    id="FileUploadForm-input"
    multiple=""
    name="formFile"
    role="input"
    type="file"
  />
</div>
`;
